src/engine/Operation.h

```diff
@@ -208,18 +213,26 @@ class Operation {
     return _warnings;
   }
 
-  void checkAbortion() const;
-  // Check if the abortion flag has been set and throws an exception if that's
-  // the case. This will be called at strategic places on code that potentially
-  // can take a (too) long time. The `detailSupplier` allows to pass a message
-  // to add to any potential exception that might be thrown.
-  void checkAbortion(const std::invocable auto& detailSupplier) const;
+  AD_ALWAYS_INLINE void checkCancellation() const {
```

Comment:
Putt this function below the "general case", then you benefit from the comment, and can just coomment that this is the default version.